hypothesis-python/src/hypothesis/strategies.py

```diff
@@ -440,6 +453,19 @@ def floats(
         max_value = float_of(max_value, width)
         assert isinstance(max_value, float)
 
+    if min_value != min_arg:
+        note_deprecation(
+            'min_value=%r cannot be exactly represented as a float of width '
+            '%d, which will be an error in a future version.'
```

Comment:
Trying `min_value=1.8, width=32` made me realise that this message might not be helpful for some users.

- `18/10` is not exactly representable as an IEEE 754 float of *any* width
- Python helpfully interprets `1.8` as the nearest *64-bit* float, and for consistency the repr of that value is `'1.8'`.  This hides the inexactness from unwary users.
- All is still well *until* someone specifies a narrower width, which is clearly within the range but can be more closely approximated by 64-bit floats.

TLDR; the message should end `...future version.  Use min_value=%r instead.' % (min_value, width, min_arg)`

No need to do this for integers though as the nearest integer is immediately obvious.